### DISCUSSION TITLE
About undefined keys in keypad mode

### DISCUSSION THREAD

**Discussion by norris-young** [meow-edit/meow#271] [Q&A] [answered]
If keypad mode is designed for executing commands, with a leader keymap(as shown below) without defining `a`, I think it should do nothing. However, a `SPC a` inserts a letter `a` in **normal** state instead. I really feel confused about this behaviour. Is there any variable to disable it?

```emacs-lisp
(meow-leader-define-key
  ;; SPC j/k will run the original command in MOTION state.
  '("j" . "H-j")
  '("k" . "H-k")
  ;; Use SPC (0-9) for digit arguments.
  '("1" . meow-digit-argument)
  '("2" . meow-digit-argument)
  '("3" . meow-digit-argument)
  '("4" . meow-digit-argument)
  '("5" . meow-digit-argument)
  '("6" . meow-digit-argument)
  '("7" . meow-digit-argument)
  '("8" . meow-digit-argument)
  '("9" . meow-digit-argument)
  '("0" . meow-digit-argument)
  '("/" . meow-keypad-describe-key)
  '("?" . meow-cheatsheet))
```

**Comment by eshrh** [comment, marked as answer]
I think this is a pretty reasonable request. Please rebuild your meow on the latest commit, because in dcc705b7aba3b7a194fb7f44923ab4a6ef0e0cf9 I added the variable `meow-keypad-self-insert-undefined`, which you would want to set to nil.

I don't think that this is good for most users though, so I erred on the side of caution and left the default value to `t`, so this does not change default behavior.

**Reply by norris-young** [reply]
Great! Thanks:D

**Reply by fbergroth** [reply]
> I don't think that this is good for most users though, so I erred on the side of caution and left the default value to `t`, so this does not change default behavior.

I'm curious, in what scenarios do people want the self-insert fallback?

**Reply by eshrh** [reply]
Honestly now that you make me think about, I couldn't tell you. I guess I just accepted how it works and took it for granted. Do you think we should set the default to nil? Or do you think we should remove the option and make this the only behavior?

The reason this happens in the first place isn't a sparse/normal keymap issue like you might expect, but rather that the leader internally looks up keys under the standard `C-c` mode-specific-map (and also adds dispatch commands there as well). Maybe this was unintentional?

**Reply by fbergroth** [reply]
I can't think of any use case for the self-insert fallback either. 

I like to keep options to a minimum if possible, so I think we should remove it unless we find a compelling reason to keep it.

**Reply by DogLooksGood** [reply]
Well, in motion state. If you bind `x` to a command, and `SPC x` will execute the original command on `x`? I mean, without any setup.

**Reply by norris-young** [reply]
Unless you bind `x` to `H-x` with meow-leader-define-key, it will still do self-insert.
And in my cases, motion state is usually used in read-only buffers, so I will choose the binding for original command on `x` instead of this self-insert.
